### ISSUE TITLE
Error when downloading blobs with storage v1.1.0

### ISSUE THREAD

**Issue by jonparrott** [GoogleCloudPlatform/google-cloud-python#3342]
See build log: https://www.google.com/url?q=https%3A%2F%2Ftravis-ci.org%2FGoogleCloudPlatform%2Fpython-docs-samples%2Fbuilds%2F226935178%23L786&sa=D&sntz=1&usg=AFQjCNG3mWBi07AXjvj70yKUdQ0pI4hJNw

**Comment by dhermes** [comment]
So this is kind of a bug, kind of not a bug, but **some code** needs to change.

TLDR; **My vote for a fix**: make the `Blob.download_to_filename` [file timestamp update][4] optional, dependent on `self.updated is not None`

----

The [old implementation][1] did something somewhat silly. It [called `reload`][2] on the instance if the `media_link` wasn't present, essentially forcing users to incur two round trips.

I "turned off" this "feature" in my fix (and @tseaver [asked][3] why / if that was OK during code review). However, that revealed a (IMO long-standing) bug [elsewhere in the code][4]:

```python
mtime = time.mktime(self.updated.timetuple())
```

i.e. the `Blob.download_to_filename` **assumes** the `updated` property is populated, which **assumes** `reload` has been called (or the `_properties` have been populated in some other way).


[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/storage-1.0.0/storage/google/cloud/storage/blob.py#L349-L371
[2]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/storage-1.0.0/storage/google/cloud/storage/blob.py#L350-L351
[3]: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/3323#discussion_r113035901
[4]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/storage-1.1.0/storage/google/cloud/storage/blob.py#L435

**Comment by jonparrott** [comment]
> TLDR; My vote for a fix: make the Blob.download_to_filename file timestamp update optional, dependent on self.updated is not None

Fine with me.

**Comment by danigosa** [comment]
Same issue here, is there any workaround or sticking to 1.0.0 'til the fix? All our tests failed miserably because of the upgrade.

**Comment by dhermes** [comment]
@danigosa I'll send out a fix today and we can push a `1.1.1` release. SGTY @jonparrott and @lukesneeringer?

**Comment by jonparrott** [comment]
Yes
